utilities/src/main/kotlin/org/jetbrains/kotlin/cli/utilities/GeneratePlatformLibraries.kt

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package org.jetbrains.kotlin.cli.utilities
+
+import org.jetbrains.kotlin.cli.bc.K2Native
+import org.jetbrains.kotlin.konan.file.File
+import java.util.concurrent.*
+import kotlinx.cli.*
+
+fun generatePlatformLibraries(args: Array<String>) {
+    val argParser = ArgParser("generate-platform")
+    val inputDirectoryPath by argParser.option(
+            ArgType.String, "input-directory", "i", "Input directory").required()
+    val outputDirectoryPath by argParser.option(
+            ArgType.String, "output-directory", "o", "Output directory").required()
+    val target by argParser.option(
+            ArgType.String, "target", "t", "Compilation target").required()
+    val saveTemps by argParser.argument(
```

Comment:
Why `argument`, not `option` ?

Reply:
Option has value, argument - does not.

Reply:
IMHO in this case a lot more convenient is optional option `--save-temps`
If present - save, if not - delete.

Reply:
As far as I understand, argument is something passed without a flag at all. So currently the tools saves temporary data if the following args are passed:
```
./generate-platform --input-directory ... --output-directory ... --target macos_x64 true
```

Reply:
For me it works as `generate-platform  ... --target mingw save-temps`

Reply:
OK, reworked here a bit, indeed boolean option seems to be better fit.